src/App.tsx

```diff
@@ -1,17 +1,98 @@
-import React from 'react';
-import './App.css';
-
+import React, { ChangeEvent, useState } from 'react';
 import users from './api/users';
+import todos from './api/todos';
+import { TodoList } from './components/TodoList/TodoList';
+import { PreparedTodo } from './react-app-env';
+
+const preparedTodos: PreparedTodo[] = todos.map((todo) => {
+  const user = users.find(person => person.id === todo.userId) || undefined;
+
+  return (
+    {
+      ...todo,
+      user,
+    }
+  );
+});
 
 const App: React.FC = () => {
+  const [todoList, setTodoList] = useState(preparedTodos);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [todoUser, setTodoUser] = useState(0);
+  const [showTitleError, setShowTitleError] = useState(false);
+  const [showUserError, setShowUserError] = useState(false);
+
+  const addTodo = (title: string, userId: number):void => {
+    const todo: PreparedTodo = {
+      userId,
+      id: todoList.length + 1,
```

Comment:
> - `id` is a maximal `id` in the array + 1

not `list.length`
something like this:
`Math.max(...array.map((todo) => todo.id)) + 1`